### ISSUE TITLE
Subscribing to a subscription from the client side.

### ISSUE THREAD

**Issue by jamessmith-decisionlab** [tartiflette/tartiflette#335]
Hi,

I have implemented subscriptions on the server side as per your getting started tutorial. On the relevant page...

https://tartiflette.io/docs/tutorial/running-a-subscription

...I see that you do not provide the code to subscribe to the subscription using Python, you simply subscribe to it via the GraphiQL client. However, on this page...

https://tartiflette.io/docs/api/subscription

...it appears that you do indeed provide the relevant code. I've implemented this is a separate application, thus:

```
    async def readSensorValues() -> None:
        engine = await create_engine("./app/sdl")

        query = """
        
            subscription {
              readSensorValues ( pmsId: "1" ) {
                pmsId
                dataValue
              }
            }
        
        """

        async for result in engine.subscribe(query=query):
            print(result)
```

The `./app/sdl` directory contains the identical schema file to the one that is being used by the server to implement the subscription.

Now I need some way to connect this engine to an `aiohttp` client session somehow? But there is no code example to show how this is done. 

Can you please provide something, assuming it's possible?

**Comment by abusi** [comment]
I'm not sure that I understand your problem. Can you be more specific about what you're trying to achieve ?

Today subscriptions in `tartiflette-aiohttp` support WebSocket only using Aiohttp Websocket classes. (https://github.com/tartiflette/tartiflette-aiohttp/blob/master/tartiflette_aiohttp/_subscription_ws_handler.py).

The subscriber should then implement websocket protocols. 

If you want to support another communication protocol, feel free to integrate `tartiflette` (not `tartiflette-aiohttp`) into anything you want.
The only requierment for subscription in `tartiflette` is that is should be called in an async generator. 

Support on `tartiflette-aiohttp` for anything other websocket is not planned but if you wanna do it, feel free to PR :).

But, i'm still not sure I understand want you want to do :/

**Comment by jamessmith-decisionlab** [comment]
Hi, thanks for the prompt response.

It is difficult to find the words but let me try again.

Your getting started tutorial explains how to set up subscriptions on the *server* side. I want to set up subscriptions on the *client* side.

At the moment, I can only subscribe to the subscriptions via the GraphiQL client. I want to subscribe to the subscriptions *programmatically* using your library.

The code you have here...

https://tartiflette.io/docs/api/subscription

...suggests that this is possible, but does not provide all the details.

**Comment by jamessmith-decisionlab** [comment]
I would expect something like:

```
import web from aiohttp

client = web.ClientSession(url)

client.registerGraphQLSubscriptionHandler(readSensorValues)

client.post("""
        
            subscription {
              readSensorValues ( pmsId: "1" ) {
                pmsId
                dataValue
              }
            }
        
        """)

```
Do you see what I mean?

**Comment by jamessmith-decisionlab** [comment]
Here is something that comes close...

https://github.com/ecthiender/py-graphql-client

...but it doesn't support HTTP, or HTTPS for that matter. I would like to use your equivalent.

**Comment by Maximilien-R** [comment]
Hi @jamessmith-decisionlab,

If I do correctly understand your need, you would like to hit your GraphQL service through another Python service (or script) programmatically?

If so, IMO, it's a client issue and not related to Tartiflette itself. We don't provide client utilities at the moment and we don't plan to since we unfortunatly don't have enough time to work on such things like a Python client supporting websocket using the [`apollo-transport-ws`](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md).

[`py-graphql-client`](https://github.com/ecthiender/py-graphql-client) seems to already do it. If it doesn't support HTTP neither HTTPS, you could still use another Python HTTP client for this and use `py-graphql-client` only for subscriptions over websocket.

Also, you can take a look on the [`aiohttp`](https://docs.aiohttp.org/en/stable/client_quickstart.html#websockets) library which provides websockets support out-of-the-box, but you'll have to implements the [`apollo-transport-ws`](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md) protocol.

**Comment by abusi** [comment]
@jamessmith-decisionlab we currently do not provide client side librairy as our lib (`tartfilette`) is integretion independant (meaning it's the user that choose the web server and protocol, http is not even mandatory, `tartiflette` is able to be under a gRPC server for example).

If you think that is something missing to our universe, feel free to create one and we'll gladly add it to the organization.

**Comment by jamessmith-decisionlab** [comment]
Hi, thanks for the clarification.

It seems to me you have already started client side support, as the first listing on this page testifies:

https://tartiflette.io/docs/api/subscription

I definitely think you should consider finishing it.

In a meantime, however, I will try out https://github.com/ecthiender/py-graphql-client.

**Comment by jamessmith-decisionlab** [comment]
Many thanks for responding to all my questions lately, by the way!
